File: app/src/main/java/com/desafioconcret/net/GitHubApi.java
package com.desafioconcret.net;

import com.desafioconcret.pojo.json.PullRequests;
import com.desafioconcret.pojo.json.TopRepositorios;

import java.util.List;

import io.reactivex.Observable;

import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface GitHubApi {

    @GET("search/repositories?")
    Observable<TopRepositorios> getTopRepositorios(@Query("q") String q, @Query("sort") String sort, @Query("page") Integer page);
//    Call<TopRepositorios> getTopRepositorios (@Query("q") String q, @Query("sort") String sort, @Query("page") Integer page);


    @GET("repos/{criador}/{repositorio}/pulls")
    Observable<List<PullRequests>> getPulls(@Path("criador") String creator, @Path("repositorio") String repositorie);


}

File: app/src/main/java/com/rafaelmfer/consultinggithub/retrofit/GitHubApi.kt
package com.rafaelmfer.consultinggithub.retrofit

import com.rafaelmfer.consultinggithub.mvvm.model.pullrequests.GitPullRequestResponse
import com.rafaelmfer.consultinggithub.mvvm.model.repositories.GitRepositoriesResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GitHubApi {

    @GET("search/repositories?q=language:Java&sort=stars")
    fun getRepositoriesList(
        @Query("page") page: Int
    ): Call<GitRepositoriesResponse>

    @GET("repos/{creator}/{repository}/pulls")
    fun getPullRequests(
        @Path("creator") creatorString: String,
        @Path("repository") repoString: String,
        @Query("page") page: Int
    ): Call<List<GitPullRequestResponse>>
}